source/core_mqtt_serializer.c

```diff
@@ -2623,6 +2628,33 @@ MQTTStatus_t MQTT_GetIncomingPacketTypeAndLength( TransportRecv_t readFunc,
 
 /*-----------------------------------------------------------*/
 
+MQTTStatus_t MQTT_UpdateDuplicatePublishFlag( uint8_t * pHeader,
+                                              bool set )
+{
+    MQTTStatus_t status = MQTTSuccess;
+
+    if( pHeader == NULL )
+    {
+        status = MQTTBadParameter;
+    }
+    else if( ( ( *pHeader ) & MQTT_PACKET_TYPE_PUBLISH ) == 0 )
```

Comment:
Is this correct? Should it not be `((*pHeader) & 0xF0U) != MQTT_PACKET_TYPE_PUBLISH`?

Reply:
We are just verifying whether the publish bit is set. Other bits can be set as well.

I think this should suffice.